### PR TITLE
fix NestedValue's Map type

### PR DESCRIPTION
**Pull request by bluebill1049**

### CODE REVIEW

File: src/types/form.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ import { FieldArrayDefaultValues } from './fieldArray';
 declare const $NestedValue: unique symbol;
 
 export type NestedValue<
-  TValue extends unknown[] | Record<string, unknown> =
+  TValue extends unknown[] | Record<string, unknown> | Map<string, any> =
```
Comment (kotarella1110) [inline src/types/form.ts:22]: Do we need to support `Set` as well?

Comment (bluebill1049) [reply src/types/form.ts:22]: probably until someone reports another issue?

**Review by kotarella1110 — APPROVED** [review]
LGTM 🎉

**Comment by bluebill1049** [comment]
thank you for the review @kotarella1110 👍

File: src/types/form.ts (hunk 1 above)
Comment (amcsi) [reply src/types/form.ts:22]: Also I don't understand the need to restrict the keys of Map to string, when NestedValue means the Map is to be treated as a leaf for the form value object tree. Shouldn't it be `Map<unknown, unknown>` or `Map<any, any>` ?

**Pull request merged**